next.config.js

```diff
@@ -3,12 +3,41 @@ const withCSS = require('@zeit/next-css')
 const withSourceMaps = require('@zeit/next-source-maps')
 const optimizedImages = require('next-optimized-images')
 const withOffline = require('next-offline')
-
 const webpackExtra = require('./webpack.extra')
+const webpack = require('webpack')
+require('dotenv').config()
 
 const nextConfiguration = {
-  webpack: config => Object.assign({}, config, webpackExtra),
+  webpack: (config, options) => {
+    const entryFactory = config.entry
+    const { isServer } = options
+    config.plugins.push(new webpack.EnvironmentPlugin(process.env))
+    if (!isServer) {
+      const newConfig = {
+        ...config,
+        entry: () =>
+          entryFactory().then(entry => {
+            const main = entry['main.js']
+            main.push('./pages/_document.js')
```

Comment:
We'll need to list all pages from the application here? There isn't a way to include all pages automatically?